tracer/src/Datadog.Trace/Baggage.cs

```diff
@@ -0,0 +1,398 @@
+﻿// <copyright file="Baggage.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System.Collections;
+using System.Collections.Generic;
+using System.Threading;
+
+#nullable enable
+
+namespace Datadog.Trace;
```

Comment:
Baggage class is not in the Propagators namespace as well?

Reply:
I don't think so. To me `Baggage` as only the collection of baggage items, which has no knowledge of how it's propagated across services. It can even be used _without_ propagation, for example, by adding baggage at one point early during a transaction and reading it back later from the same service.

In the next PR I'm adding the baggage propagator, which is definitely in the `Propagators` namespace, along with the trace context propagators.